### PR TITLE
Migrate protocol-http package to @smithy/protocol-http

### PR DESCRIPTION
**Pull request by srchase**
This PR migrates the protocol-http package from https://github.com/aws/aws-sdk-js-v3/tree/main/packages/protocol-http, moving it to the @smithy NPM scope. It pulls in the relevant git history from [aws-sdk-js-v3](https://github.com/aws/aws-sdk-js-v3).

Additionally, it adds a `@smithy/types`, which contains the types used by `@smithy/protocol-http`, along with the types used by the SSDK libs. The SSDK libs are updated to use the `@smithy` scoped packages.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-ssdk-libs/server-node/src/node.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { HttpRequest, HttpResponse } from "@aws-sdk/protocol-http";
-import { HeaderBag, QueryParameterBag } from "@aws-sdk/types";
+import { HeaderBag, HttpRequest, HttpResponse } from "@smithy/protocol-http";
```
[the remark this answers is not in the document]
Comment (srchase) [reply smithy-typescript-ssdk-libs/server-node/src/node.ts:6]: I explicitly moved `HeaderBag` from `@aws-sdk/types` to `@smithy/protocol-http`. I tried to pull over as many of the types into protocol-http itself, but the SSDK libs rely on `SerdeContext`, which I didn't think made sense to pull into protocol-http.

File: packages/protocol-http/package.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "clean": "rimraf ./dist-* && rimraf *.tsbuildinfo",
+    "lint": "eslint -c ../../.eslintrc.js \"src/**/*.ts\"",
+    "format": "prettier --config ../../prettier.config.js --ignore-path ../.prettierignore --write \"**/*.{ts,md,json}\"",
+    "test": "jest"
+  },
+  "main": "./dist-cjs/index.js",
+  "module": "./dist-es/index.js",
+  "types": "./dist-types/index.d.ts",
+  "author": {
+    "name": "AWS SDK for JavaScript Team",
```
[the remark this answers is not in the document]
Comment (srchase) [reply packages/protocol-http/package.json]: Changed to Smithy team. The Changelog points back to the [aws-sdk-js-v3](https://github.com/aws/aws-sdk-js-v3) repository.

**Review by pose — APPROVED** [review]

File: packages/protocol-http/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * header should be represented as a single string with values separated by
+ * `, `.
+ *
+ * Keys should be considered case insensitive, even if this is not enforced by a
+ * particular implementation. For example, given the following HeaderBag, where
+ * keys differ only in case:
+ *
+ * ```json
+ *    {
+ *        'x-amz-date': '2000-01-01T00:00:00Z',
```
Comment (hpmellema) [inline packages/protocol-http/src/types.ts:16]: Should we be updating these to pull out amazon-specific documentation like this? (i.e `x-request-date` vs `x-amz-date`?

Comment (srchase) [reply packages/protocol-http/src/types.ts]: Yeah, no point in leaving that as `amz`.

File: packages/protocol-http/src/Fields.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Field } from "./Field";
+import { FieldPosition } from "./FieldPosition";
+
+export type FieldsOptions = { fields?: Field[]; encoding?: string };
```
[the remark this answers is not in the document]
Comment (srchase) [reply packages/protocol-http/src/Fields.ts:4]: Moved it over, and the above.

**Review by hpmellema — APPROVED** [review]

**Pull request merged**